File: javascript/prime.js
function sieve(size) {
    if (size <= 0) return [];
    if (size == 1) return [false];
    var sieveArray = new Array(size);
    sieveArray[0] = sieveArray[1] = false;
    for (var i = 2; i < size; ++i) sieveArray[i] = true;
    var root = (Math.sqrt(size) | 0) + 1;
    for (var i = 2; i < root; ++i) {
        if (!sieveArray[i]) continue;
        for (var j = i * i; j < sieveArray.length; j += i) {
            sieveArray[j] = false;
        }
    }
    return sieveArray;
}

function countPrime(n) {
    var sieveArray = sieve(n + 1);
    var count = 0;
    for (var i = 0; i < sieveArray.length; ++i) {
        if (sieveArray[i]) ++count;
    }
    return count;
}

var n = (process.argv.length >= 3) ? parseInt(process.argv[2]) : 10000000;
console.log(countPrime(n));

File: c/prime.c
#include <math.h>
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

bool* sieve(int size) {
    bool* sieveArray = calloc(size, sizeof(bool));
    for (int i = 2; i < size; ++i) { sieveArray[i] = true; }
    int root = sqrt(size) + 1;
    for (int i = 2; i < root; ++i) {
        if (!sieveArray[i]) { continue; }
        for (int j = i * i; j < size; j += i) {
            sieveArray[j] = false;
        }
    }
    return sieveArray;
}

int countPrime(int n) {
    bool* sieveArray = sieve(n + 1);
    int count = 0;
    for (int i = 0; i <= n; ++i) {
        if (sieveArray[i]) { ++count; }
    }
    free(sieveArray);
    return count;
}

int main(int argc, char* argv[]) {
    int n = (argc >= 2) ? strtol(argv[1], NULL, 10) : 10000000;
    printf("%d\n", countPrime(n));
}
